### PR TITLE
Move to SPM compatible layout

### PR DESCRIPTION
**Pull request by timshadel**
The basic checks for Cocoapods and SPM pass:

```
$ pod lib lint

 -> Swiftification (9.2.1)

Swiftification passed validation.
```

```
$ swift build
Compile Swift Module 'Swiftification' (10 sources)
```

```
$ swift test
Compile Swift Module 'SwiftificationTests' (10 sources)
/Users/timshadel/projects/open/Swiftification/Tests/SwiftificationTests/ObserverSetTests.swift:93:13: warning: variable 'obj' was written to, but never read
        var obj: TestObserver? = TestObserver(observee: observee)
            ^
Linking ./.build/debug/SwiftificationPackageTests.xctest/Contents/MacOS/SwiftificationPackageTests
Test Suite 'All tests' started at 2017-01-17 10:25:12.094
Test Suite 'SwiftificationPackageTests.xctest' started at 2017-01-17 10:25:12.095
[...]
Test Suite 'SwiftificationPackageTests.xctest' passed at 2017-01-17 10:25:13.128.
	 Executed 68 tests, with 0 failures (0 unexpected) in 1.031 (1.033) seconds
Test Suite 'All tests' passed at 2017-01-17 10:25:13.128.
	 Executed 68 tests, with 0 failures (0 unexpected) in 1.031 (1.034) seconds
```

The project was regenerated using SPM, and the build and all the tests pass in Xcode. If there were special build settings in the `xcodeproj` or the (now removed) `xcworkspace` then I need to go back and fix something. I haven't tested pulling it in as a dependency with Cocoapods, though I think you could do that with `pod 'Swiftification', :git => 'https://github.com/timshadel/Swiftification.git'` to be sure.

### CODE REVIEW

**Comment by hiltonc** [comment]
Awesome! You may need to update `.travis.yml` so that unit tests pass on Travis (weird that they haven't even started yet).

File: Swiftification.xcodeproj/xcshareddata/xcschemes/Swiftification.xcscheme [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <Scheme
-   LastUpgradeVersion = "0800"
+   LastUpgradeVersion = "9999"
```
Comment (nickmshelley) [inline Swiftification.xcodeproj/xcshareddata/xcschemes/Swiftification.xcscheme]: This seems odd.

File: Swiftification.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -7,479 +7,411 @@
...
-				D96A72D71C089B5C002CCCEE /* Sequence.swift */,
-				D45479CA1C46D336007258A6 /* String.swift */,
-				D96A72BF1C089B12002CCCEE /* Swiftification.h */,
+				OBJ_6 /* Package.swift */,
+				OBJ_7 /* Sources */,
+				OBJ_19 /* Tests */,
+				OBJ_31 /* Products */,
 			);
-			path = Swiftification;
+			name = "";
```
Comment (nickmshelley) [inline Swiftification.xcodeproj/project.pbxproj]: Is this a group with an empty name?

File: Swiftification.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -7,479 +7,411 @@
...
-					D96A72BB1C089B12002CCCEE = {
-						CreatedOnToolsVersion = 7.1.1;
-						LastSwiftMigration = 0800;
-					};
-					D96A72C51C089B13002CCCEE = {
-						CreatedOnToolsVersion = 7.1.1;
-						LastSwiftMigration = 0800;
-					};
-				};
+				LastUpgradeCheck = 9999;
```
Comment (nickmshelley) [inline Swiftification.xcodeproj/project.pbxproj]: Here's 9999 again...

File: Swiftification.xcodeproj/xcshareddata/xcschemes/Swiftification.xcscheme (hunk 1 above)
Comment (hiltonc) [inline Swiftification.xcodeproj/xcshareddata/xcschemes/Swiftification.xcscheme]: With the workspace deleted, I'm guessing you can delete the scheme as well?

**Comment by nickmshelley** [comment]
I noticed some weird things with the project/workspace files. Did you make this change with AppCode or something?

File: Swiftification.xcodeproj/xcshareddata/xcschemes/Swiftification.xcscheme (hunk 1 above)
Comment (timshadel) [inline Swiftification.xcodeproj/xcshareddata/xcschemes/Swiftification.xcscheme]: Schemes _do_ exist at merely the project level, apparently. (I checked after your previous comment)

**Comment by timshadel** [comment]
No, it's made with `swift package generate-xcodeproj`.

**Comment by nickmshelley** [comment]
Interesting. I'm worried that might cause problems with future versions of Xcode since they may just do a compare on that and skip the upgrade check.

**Pull request closed without merging**